front/src/applications/operationalStudies/components/MacroEditor/NGE.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef } from 'react';
+
+/* eslint-disable import/extensions, import/no-unresolved */
+import ngeMain from '@osrd-project/netzgrafik-frontend/dist/netzgrafik-frontend/en/main.js?url';
+import ngePolyfills from '@osrd-project/netzgrafik-frontend/dist/netzgrafik-frontend/en/polyfills.js?url';
+import ngeRuntime from '@osrd-project/netzgrafik-frontend/dist/netzgrafik-frontend/en/runtime.js?url';
+import ngeStyles from '@osrd-project/netzgrafik-frontend/dist/netzgrafik-frontend/en/styles.css?url';
+import ngeVendor from '@osrd-project/netzgrafik-frontend/dist/netzgrafik-frontend/en/vendor.js?url';
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+/* eslint-enable import/extensions, import/no-unresolved */
+
+import { ngeCreateOperation, ngeUpdateOperation } from './ngeOperations';
+import type {
+  NetzgrafikDto,
+  NGEDeleteEvent,
+  NGEUpdateEvent,
+  TrainrunSectionOperation,
+} from './types';
+
+interface NGEElement extends HTMLElement {
+  netzgrafikDto: NetzgrafikDto;
+}
+
+const frameSrc = `
+<!DOCTYPE html>
+<html class="sbb-lean sbb-light">
+  <head>
+    <link rel="stylesheet" href="${ngeStyles}"></link>
+    <script type="module" src="${ngeRuntime}"></script>
+    <script type="module" src="${ngePolyfills}"></script>
+    <script type="module" src="${ngeVendor}"></script>
+    <script type="module" src="${ngeMain}"></script>
+  </head>
+  <body></body>
+</html>
+`;
+
+const NGE = ({ dto }: { dto?: NetzgrafikDto }) => {
+  const frameRef = useRef<HTMLIFrameElement>(null);
+
+  let ngeRoot: NGEElement | null = null;
+
+  const [postTrainSchedule] =
+    osrdEditoastApi.endpoints.postV2TimetableByIdTrainSchedule.useMutation();
+  const [deleteTrainSchedules] = osrdEditoastApi.endpoints.deleteV2TrainSchedule.useMutation();
+  const [putV2TimetableByIdTrainSchedule] =
+    osrdEditoastApi.endpoints.putV2TrainScheduleById.useMutation();
+
+  const { getTimetableID } = useOsrdConfSelectors();
+  const timetableId = useSelector(getTimetableID);
+
+  const createTrainSchedule = async (trainrunSection: TrainrunSectionOperation) => {
+    await postTrainSchedule({
+      id: timetableId!,
+      body: ngeCreateOperation(trainrunSection),
+    }).unwrap();
+  };
+
+  useEffect(() => {
+    const frame = frameRef.current!;
+
+    const handleFrameLoad = () => {
+      frame.removeEventListener('load', handleFrameLoad);
+
+      ngeRoot = frame.contentDocument!.createElement('sbb-root') as NGEElement;
+      frame.contentDocument!.body.appendChild(ngeRoot);
+
+      // listens to create, update and delete operations
+      ngeRoot.addEventListener('operation', async (event: Event) => {
```

Comment:
Maybe we could introduce a function which takes an NGE operation and handles it inside `ngeOperations.ts`? Basically extract this callback and move it to that file, since it's specific to the NGE → OSRD conversion.

Reply:
Done